storybookjs/storybook
# Build: Move monorepo to Yarn 2

Applying https://github.com/storybookjs/storybook/pull/13559 step by step to not face gigantic conflicts or issues with the webpack 4/5 works.

## What I did

> Upgrading the Yarn binary to the latest stable release (2.4), which should make them faster.
> 
> I've configured the project to stay on the node_modules linker since it makes the upgrade fairly painless. Tests pass locally, I'll see how it goes on CI and tweak the command line references accordingly (things like --frozen-lockfile being renamed to --immutable, etc).
>
> Notable changes:

> The Yarn binary is checked within this repository. This means that all your contributors will now be ensured to use the exact same version of Yarn, regardless of what they have installed globally.
>
> The root package.json now lists all workspaces as dependencies (using the workspace: protocol). This is because in Yarn 2 the root workspace doesn't implicitly depend on all workspaces anymore, they have to be explicitly listed. In theory, the dependencies should be listed in each workspace that depends on them (so, for example, ui should depend on api) but I'm worried it would lead to a gigantic diff, so, for now, the repo keeps relying on the hoisting to work.

## How to test

- CI should be 🟢 
- To test things locally, you should be able to do the following: 
  - Checkout this branch
  - `yarn install`
  - `yarn bootstrap --core`
  - `yarn test`
  - Go in any example and run `yarn storybook`
